src/components/rb_search_widget.tsx

```diff
@@ -0,0 +1,50 @@
+import React, {useState} from "react";
+
+const formatter = new Intl.NumberFormat('en-US', {
+    style: 'currency',
+    currency: 'USD',
+    minimumFractionDigits: 2
+});
+
+export function RBSearchWidget(props) {
```

Comment:
A good practice is to start adding interfaces from the beginning, so you save some refactoring in the future!
```
 interface IRBSearchWidget {
    state: string;
    city: string;
    zipcode: number;
    newTab: boolean;
} 
```

And if you want to separate control props from UI props you can do
```
 interface IRBSearchWidget {
    settings: {
        state: string;
        city: string;
        zipcode: number;
    };
    newTab: boolean;
} 
```